src/EnergyPlus/DaylightingManager.cc

```diff
@@ -6794,58 +6695,55 @@ void DayltgInteriorIllum(EnergyPlusData &state, int &ZoneNum) // Zone number
 
     // Calculate glare index at each reference point assuming the daylight illuminance setpoint is
     //  met at both reference points, either by daylight or electric lights
-    int const enclNum = state.dataHeatBal->Zone(ZoneNum).zoneFirstSpaceSolEnclosure;
     for (int IL = 1; IL <= NREFPT; ++IL) {
         BACL = max(SetPnt(IL) * state.dataDaylightingData->enclDaylight(enclNum).aveVisDiffReflect / DataGlobalConstants::Pi,
-                   state.dataDaylightingData->ZoneDaylight(ZoneNum).BacLum(IL));
+                   thisDaylightControl.BacLum(IL));
         // DayltgGlare uses ZoneDaylight(ZoneNum)%SourceLumFromWinAtRefPt(IL,1,loop) for unshaded windows, and
         //  ZoneDaylight(ZoneNum)%SourceLumFromWinAtRefPt(IL,2,loop) for shaded windows
-        DayltgGlare(state, IL, BACL, GLRNDX(IL), ZoneNum);
+        DayltgGlare(state, IL, BACL, GLRNDX(IL), daylightCtrlNum);
     }
 
     // Check if glare level is less than maximum allowed at each ref pt.  If maximum
     // is exceeded at either ref pt, attempt to reduce glare to acceptable level by closing
     // shading device on windows that have shades that have not already been closed.
     GlareFlag = false;
     for (int IL = 1; IL <= NREFPT; ++IL) {
-        if (GLRNDX(IL) > state.dataDaylightingData->ZoneDaylight(ZoneNum).MaxGlareallowed) {
+        if (GLRNDX(IL) > thisDaylightControl.MaxGlareallowed) {
             GlareFlag = true;
             break;
         }
     }
 
-    Array3D<Real64> WDAYIL(
+    Array3D<Real64> WDAYIL(2,
```

Comment:
Heap allocated should preferably not be local variables unless the function is only called once, e.g., during initialization or final reporting then it doesn't matter. If the function is called every time step then these should be moved to `state`.

Reply:
#9067 (which builds on this branch) moves these arrays to state.